lib/Doctrine/ORM/Cache/DefaultCacheFactory.php

```diff
@@ -178,6 +181,19 @@ public function buildQueryCache(EntityManagerInterface $em, $regionName = null)
      */
     public function buildCollectionHydrator(EntityManagerInterface $em, array $mapping)
     {
+        if ($mapping['cache']) {
+            $targetPersister = $em->getUnitOfWork()->getEntityPersister($mapping['targetEntity']);
+
+            if ($targetPersister instanceof CachedPersister) {
+
+                $targetRegion = $targetPersister->getCacheRegion();
+
+                if ($targetRegion instanceof MultiGetRegion) {
+                    return new MultiGetCollectionHydrator($em);
```

Comment:
I have "fixed" this problem with https://github.com/goetas/doctrine2/commit/e5ee53807581c52d952c785568cb594826c29b8f but i think that is not the right solution...